rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java

```diff
@@ -2167,19 +2110,21 @@ public TypeTree visitVariableExpressionType(VariableExpression expression) {
 
         @Override
         public void visitVariableExpression(VariableExpression expression) {
-            JavaType type;
-            if (expression.isDynamicTyped() && expression.getAccessedVariable() != null && expression.getAccessedVariable().getType() != expression.getOriginType()) {
-                type = typeMapping.type(staticType(expression.getAccessedVariable()));
-            } else {
-                type = typeMapping.type(staticType((org.codehaus.groovy.ast.expr.Expression) expression));
-            }
-            queue.add(new J.Identifier(randomId(),
-                    sourceBefore(expression.getName()),
-                    Markers.EMPTY,
-                    emptyList(),
-                    expression.getName(),
-                    type, null)
-            );
+            queue.add(insideParentheses(expression, fmt -> {
```

Comment:
This was just a bug in the parser, the visitVariableExpression should also use the `visitVariableExpression` so things like:

```groovy
def timestamp(int hours, int minutes, int seconds) {
  (hours) * 60 * 60 + (minutes * 60) + seconds
}
```

starts working!